src/libos/crates/host-socket/src/common/sockopt.rs

```diff
@@ -0,0 +1,280 @@
+use std::convert::TryFrom;
+
+use async_io::impl_ioctl_cmd;
+use libc::{c_int, c_void, socklen_t};
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+cfg_if::cfg_if! {
+    if #[cfg(feature = "sgx")] {
+        use libc::ocall::getsockopt as do_getsockopt;
+        use libc::ocall::setsockopt as do_setsockopt;
+    } else {
+        use libc::getsockopt as do_getsockopt;
+        use libc::setsockopt as do_setsockopt;
+    }
+}
+
+use crate::prelude::*;
+
+const SOL_SOCKET: i32 = 1;
+
+pub fn new_getsockopt_cmd(
```

Comment:
A function taking raw pointers can hardly be safe (and this function is definitely not). Either mark this function as unsafe and document how to use it safely or refactor the type of `optval` to `&mut [u8]`.